xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp

```diff
@@ -11,30 +11,92 @@
 /* prototypes */
 
 real32 rasterizer_dx9_fullscreen_effect_calculate_position_z(real32 z_far, int32 a2);
-void rasterizer_dx9_fullscreen_calculate_position(const real_vector4d* location, real32 z_far, real_vector4d* output);
 
 /* public code */
 
 void rasterizer_dx9_fullscreen_passes_apply_patches(void)
 {
-    // PatchCall(Memory::GetAddress(0x262684), rasterizer_dx9_apply_gamma_and_brightness);
-    
-    // splitscreen patchy fog fix, take just the viewport portion of the main render target
-    // that is used as a texture in the patchy fog draw code shader
-    WritePointer(Memory::GetAddress(0x2774CF) + 1, rasterizer_fullscreen_effects_build_vertex_buffer_cb);
-
     // splitscreen bloom fullscreen passes fix
     WritePointer(Memory::GetAddress(0x27D8C5) + 1, rasterizer_fullscreen_effects_build_vertex_buffer_color_ctx_cb);
-    WritePointer(Memory::GetAddress(0x27D8BB) + 1, rasterizer_fullscreen_effects_build_vertex_buffer_cb);
+    WritePointer(Memory::GetAddress(0x27D8BB) + 1, rasterizer_dx9_build_default_vertex_buffer);
+    return;
+}
+
+void rasterizer_dx9_fullscreen_calculate_position(const real_vector4d* location, real32 z_far, real_vector4d* output)
+{
+    ASSERT(location);
+    ASSERT(output);
+
+    real32 z = (z_far != 0.f ? rasterizer_dx9_fullscreen_effect_calculate_position_z(z_far, 1) : 1.f);
+
+    output->i = location->i * 2.f - 1.f;    // x
+    output->j = -(location->j * 2.f - 1.f); // y
+    output->k = z;                          // z
+    output->l = 1.f;                        // scale
+    return;
+}
+
+void rasterizer_dx9_fullscreen_calculate_screen_coordinates(const real_rectangle2d* bounds, const real_point2d* location, real_point2d* output)
+{
+    ASSERT(bounds);
+    ASSERT(location);
+    ASSERT(output);
+    
+    const real32 width = bounds->x1 - bounds->x0;
+    const real32 height = bounds->y1 - bounds->y0;
+
+    output[0].u = (0.5f / width) + location->u;
+    output[0].v = (0.5f / height) + location->v;
+    output[1].u = 0.f;
+    output[1].v = 1.f;
+    return;
+}
 
+void rasterizer_dx9_fullscreen_calculate_texcoords(const real_rectangle2d* bounds, const real_point2d* location, real_point2d* output)
```

Comment:
could probably make the texture index a param here

Reply:
added parameter with default of 0